support/cmd_builder/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Command Builder
+
+#![forbid(unsafe_code)]
+
+use std::ffi::OsStr;
+use std::process::Command;
+use std::process::Stdio;
+use thiserror::Error;
+
+pub mod ps;
+
+/// Error running command
+#[derive(Error, Debug)]
+pub enum CommandError {
+    /// failed to launch command
+    #[error("failed to launch command")]
+    Launch(#[from] std::io::Error),
+    /// command exited with non-zero status
+    #[error("command exited with non-zero status ({0}): {1}")]
+    Command(std::process::ExitStatus, String),
+    /// command output is not utf-8
+    #[error("command output is not utf-8")]
+    Utf8(#[from] std::string::FromUtf8Error),
+}
+
+/// Run the PowerShell script and return the output
+pub fn run_cmd(mut cmd: Command, log_stdout: bool) -> Result<String, CommandError> {
```

Comment:
I think you should scope this crate to just powershell and not generic command stuff. I saw somewhere else that you used `run_cmd` to wrap some `hvc` command... that feels like the wrong thing to me.